modules/vault-cluster/main.tf

```diff
@@ -240,7 +244,7 @@ data "aws_iam_policy_document" "instance_role" {
 }
 
 resource "aws_s3_bucket" "vault_storage" {
-  count         = var.enable_s3_backend ? 1 : 0
+  count         = var.enable_s3_backend && var.iam_instance_profile_arn == null ? 1 : 0
```

Comment:
What does the S3 bucket have to do with the instance profile?

Reply:
Yeah good point. thank you for the catch here. going to fix it.